### PR TITLE
Update public-accessibility-logic.md for S3

### PR DESCRIPTION
**Pull request by christophetd**
related to https://github.com/DataDog/dd-source/pull/55855

Note: `bucket policy` is a domain-specific term that we should keep (instead of `bucket's policy`)

### CODE REVIEW

File: content/en/security/cloud_security_management/guide/public-accessibility-logic.md [hunk 1]
```diff
@@ -35,21 +35,10 @@ The following section details on a resource-by-resource basis the logic behind p
...
-|The bucket's policy statement allows the `s3:GetObject` permission unconditionally, with resource and principal set to `"*"`. |This defines a public policy on the bucket, meaning that cross-account access is allowed. `"*"` is a wildcard, meaning access is given to any resource and principal. |
+|The bucket policy allows the `s3:GetObject` permission unconditionally, with resource and principal set to `"*"`. |This defines a public policy on the bucket, meaning that unauthenticated access is allowed. `"*"` is a wildcard, meaning access is given to any resource and principal. |
+| Both of the bucket's `public_access_block_configuration` and the AWS account's public access block (`aws_s3_account_public_access_block`) have `restrict_public_buckets` set to `false`. | None of the bucket's or account's explicitely blocks public access, meaning that the public bucket policy takes effect. |
```
Comment (hestonhoffman) [inline content/en/security/cloud_security_management/guide/public-accessibility-logic.md:41]:
```suggestion
| Both of the bucket's `public_access_block_configuration` and the AWS account's public access block (`aws_s3_account_public_access_block`) have `restrict_public_buckets` set to `false`. | None of the buckets or accounts explicitly block public access, meaning that the public bucket policy takes effect. |
```

**Review by hestonhoffman — COMMENTED** [review]
Small grammar fix

File: content/en/security/cloud_security_management/guide/public-accessibility-logic.md (hunk 1 above)
Comment (hestonhoffman) [reply content/en/security/cloud_security_management/guide/public-accessibility-logic.md:41]: Not sure if there's a word missing here? For example "bucket's policies"

**Comment by christophetd** [comment]
Yes! Thank you

**Pull request merged**